packages/apollo-server-core/src/__tests__/logger.test.ts

```diff
@@ -1,146 +1,58 @@
-      type Query {
-        field: String!
-      }
-    `,
-    logger: loggerToUse,
-    plugins: [
-      {
-        async requestDidStart({ logger }) {
-          logger.debug(KNOWN_DEBUG_MESSAGE);
-        },
-      },
-    ],
-  });
-  await server.start();
-  await server.executeOperation({
-    query: '{ field }',
-  });
-}
+const KNOWN_DEBUG_MESSAGE = 'The server is starting.';
 
 describe('logger', () => {
-  it("works with 'winston'", async () => {
-    const sink = jest.fn();
-    const transport = new (class extends WinstonTransport {
-      constructor() {
-        super({
-          format: winston.format.json(),
-        });
-      }
-
-      override log(info: any) {
-        sink(info);
-      }
-    })();
-
-    const logger = winston.createLogger({ level: 'debug' }).add(transport);
-
-    await triggerLogMessage(logger);
-
-    expect(sink).toHaveBeenCalledWith(
-      expect.objectContaining({
-        level: LOWEST_LOG_LEVEL,
-        message: KNOWN_DEBUG_MESSAGE,
-      }),
-    );
-  });
-
-  it("works with 'bunyan'", async () => {
-    const sink = jest.fn();
-
-    // Bunyan uses streams for its logging implementations.
-    const writable = new PassThrough();
-    writable.on('data', (data) => sink(JSON.parse(data.toString())));
-
-    const logger = bunyan.createLogger({
-      name: 'test-logger-bunyan',
-      streams: [
+  it('uses internal loglevel logger by default', async () => {
+    const server = new ApolloServerBase({
+      typeDefs: gql`
+        type Query {
+          field: String!
+        }
+      `,
+      plugins: [
         {
-          level: LOWEST_LOG_LEVEL,
-          stream: writable,
+          async serverWillStart({ logger }) {
+            logger.debug(KNOWN_DEBUG_MESSAGE);
+          },
         },
       ],
     });
 
-    await triggerLogMessage(logger);
-
-    expect(sink).toHaveBeenCalledWith(
-      expect.objectContaining({
-        level: bunyan.DEBUG,
-        msg: KNOWN_DEBUG_MESSAGE,
-      }),
-    );
-  });
-
-  it("works with 'loglevel'", async () => {
-    const sink = jest.fn();
+    const defaultLogger = server['logger'] as loglevel.Logger;
+    const debugSpy = jest.spyOn(defaultLogger, 'debug');
+    await server.start();
 
-    const logger = loglevel.getLogger('test-logger-loglevel');
-    logger.methodFactory =
-      (_methodName, level): loglevel.LoggingMethod =>
-      (message) =>
-        sink({ level, message });
-
-    // The `setLevel` method must be called after overwriting `methodFactory`.
-    // This is an intentional API design pattern of the loglevel package:
-    // https://www.npmjs.com/package/loglevel#writing-plugins
-    logger.setLevel(loglevel.levels.DEBUG);
-
-    await triggerLogMessage(logger);
-
-    expect(sink).toHaveBeenCalledWith({
-      level: loglevel.levels.DEBUG,
-      message: KNOWN_DEBUG_MESSAGE,
-    });
+    expect(debugSpy).toHaveBeenCalledWith(KNOWN_DEBUG_MESSAGE);
+    expect(defaultLogger.levels).toEqual(loglevel.levels);
```

Comment:
maybe instanceof instead

Reply:
Tried, but can't. Left a comment that explains why I'm asserting this at least.